test/RoutePluginManagerFactoryTest.php

```diff
@@ -19,36 +19,44 @@
 
 class RoutePluginManagerFactoryTest extends TestCase
 {
-    public function setUp()
+    /**
+     * @var ContainerInterface|\PHPUnit\Framework\MockObject\MockObject
+     */
+    private $container;
+    /**
+     * @var RoutePluginManagerFactory
+     */
+    private $factory;
+
+    public function setUp(): void
     {
-        $this->container = $this->prophesize(ContainerInterface::class);
+        $this->container = $this->createMock(ContainerInterface::class);
         $this->factory = new RoutePluginManagerFactory();
     }
 
     public function testInvocationReturnsAPluginManager()
     {
-        $plugins = $this->factory->__invoke($this->container->reveal(), RoutePluginManager::class);
+        $plugins = $this->factory->__invoke($this->container, RoutePluginManager::class);
         $this->assertInstanceOf(RoutePluginManager::class, $plugins);
     }
 
     public function testCreateServiceReturnsAPluginManager()
     {
-        $container = $this->prophesize(ServiceLocatorInterface::class);
-        $container->willImplement(ContainerInterface::class);
```

Comment:
I am not sure what this `$container->willImplement(ContainerInterface::class);` tests? can I just remove it? then we can get rid of `Prophecy` lib otherwise I don't know how to test this.

Reply:
> I am not sure what this $container->willImplement(ContainerInterface::class); tests?

See: https://github.com/phpspec/prophecy#how-to-use-it

> ```php
> $prophecy->willExtend('stdClass');
> $prophecy->willImplement('SessionHandlerInterface');
> ```
> There are 2 interesting calls - willExtend and willImplement. The first one tells object prophecy that our object should extend specific class, the second one says that it should implement some interface.

Reply:
@froschdesign so should I add prophecy back? I don't know if PHPUnit mock library has this feature. I still don't understand if we need this here. `Laminas\ServiceManager\ServiceLocatorInterface` already implements `ContainerInterface` unless this is a special case I don't see why we need it. At least I don't know why. Can revert `prophecy` if you want.

Reply:
No, everything is fine. 👍